File: skynda_backend/src/java/me/skynda/common/interfaces/services/IVehicleReportService.java
package me.skynda.common.interfaces.services;

import me.skynda.common.dto.CreateOrUpdateResponseDto;
import me.skynda.common.dto.DeleteResponseDto;
import me.skynda.vehicle.dto.VehicleReportDto;
import org.springframework.validation.BindingResult;

import java.io.Serializable;
import java.util.List;

public interface IVehicleReportService {
    List<VehicleReportDto> getAll();

    VehicleReportDto getSingleBy(Integer id);

    CreateOrUpdateResponseDto createOrUpdate(VehicleReportDto dto, BindingResult bindingResult);

    DeleteResponseDto delete(Integer id);
    List<VehicleReportDto> getAllBy(Serializable vehicleId);
}

File: skynda_dotnet_backend/Triven.Domain/Services/IVehicleReportService.cs
using System.Collections.Generic;
using Triven.Domain.Results;
using Triven.Domain.ViewModels.Vehicle;

namespace Triven.Domain.Services
{
    public interface IVehicleReportService
    {
        ServiceResult<IEnumerable<VehicleReportViewModel>> GetAll();
        ServiceResult<VehicleReportViewModel> Get(int id);
        ServiceResult<VehicleReportViewModel> Create(int vehicleId, VehicleReportViewModel dto);
        ServiceResult<VehicleReportViewModel> Update(int vehicleId, int id, VehicleReportViewModel dto);
        ServiceResult<bool> Delete(int id);
        ServiceResult<IList<VehicleReportViewModel>> GetAllBy(int vehicleId);
    }
}